python/testmonitor/results/results.py

```diff
@@ -0,0 +1,224 @@
+        inputs=None,
+        outputs=None,
+        parameters=None,
+        status=None):
+    """
+    Creates a <see cref="StepData"/> object and
+    populates it to match the TestStand data model.
+    :param name: The test step's name.
+    :param step_type: The test step's type.
+    :param inputs: The test step's input values.
+    :param outputs: The test step's output values.
+    :param parameters: The measurement parameters.
+    :param status:
+    :return: The step data used to create a test step.
+    """
+    step_status = status if status is not None else StatusObject(status_type='RUNNING')
+
+    step_data = TestStepRequestObject(
+        step_id=None,
+        parent_id=None,
+        result_id=None,
+        children=None,
+        data=parameters,
+        data_model="TestStand",
+        name=name,
+        started_at=None,
+        status=step_status,
+        step_type=step_type,
+        total_time_in_seconds=random.uniform(0, 1) * 10,
+        inputs=inputs,
+        outputs=outputs)
+
+    return step_data
+
+
+def main():
+    # Initialize SystemLink APIs
+    results_api = ResultsApi()
+    steps_api = StepsApi()
+
+    # Set test limits
+    low_limit = 0
+    high_limit = 70
+
+    test_result = TestResultRequestObject(
+        program_name="Power Test",
+        status=StatusObject(status_type='RUNNING'),
+        system_id=None,
+        host_name=None,
+        properties=None,
+        keywords=None,
+        serial_number=str(uuid.uuid4()),
+        operator="John Smith",
+        part_number="NI-ABC-123-PWR",
+        file_ids=None,
+        started_at=None,
+        total_time_in_seconds=0.0)
+
+    create_results_request = CreateTestResultsRequest(results=[test_result])
+    response = await results_api.create_results_v2(create_results_request)
+    test_result = response.results[0]
+
+    """
+    Simulate a sweep across a range of electrical current and voltage.
+    For each value, calculate the electrical power (P=IV).
+    """
+    for current in range(0, 10):
+        # Generate a parent step to represent a sweep of voltages at a given current.
+        voltage_sweep_step_data = generate_step_data("Voltage Sweep", "SequenceCall")
+        voltage_sweep_step_data.result_id = test_result.id
+        # Create the step on the SystemLink server.
+        create_steps_request = TestStepCreateOrUpdateRequestObject(
+            steps=[voltage_sweep_step_data],
+            update_result_total_time=True)
+        response = await steps_api.create_steps_v2(create_steps_request)
+        voltage_sweep_step = response.steps[0]
+
+        for voltage in range(0, 10):
+            # Simulate obtaining a power measurement.
+            power, inputs, outputs = measure_power(current, voltage)
+
+            # Test the power measurement.
+            status = StatusObject(status_type='FAILED') if (power < low_limit or power > high_limit) else StatusObject(
+                status_type='PASSED')
```

Comment:
Since the ternary is already so long that it wraps, do you think we could use a normal `if` here? e.g.

```
if power < low_limit or power > high_limit:
    status = StatusObject(status_type='FAILED')
else:
    status = StatusObject(status_type='PASSED')
```

Reply:
This is optional for this PR, but we may want to consider running `black` over this, which will homogenize the formatting everywhere (`python -m black --line-length 120 .`). I suspect it would have changed this, but i'm not 100% sure.

(you also might need to install `black` manually, I don't think it's in the SL distribution...yet)